engine/src/main/java/org/pentaho/di/trans/steps/file/BaseFileInputFiles.java

```
@@ -0,0 +1,175 @@
+  /** Wildcard or filemask (regular expression) */
+  @Injection( name = "FILEMASK", group = "FILENAME_LINES" )
+  public String[] fileMask = {};
+
+  /** Wildcard or filemask to exclude (regular expression) */
+  @Injection( name = "EXCLUDE_FILEMASK", group = "FILENAME_LINES" )
+  public String[] excludeFileMask = {};
+
+  /** Array of boolean values as string, indicating if a file is required. */
+  @Injection( name = "FILE_REQUIRED", group = "FILENAME_LINES" )
+  public String[] fileRequired = {};
+
+  /** Array of boolean values as string, indicating if we need to fetch sub folders. */
+  @Injection( name = "INCLUDE_SUBFOLDERS", group = "FILENAME_LINES" )
+  public String[] includeSubFolders = {};
+
+  /** Are we accepting filenames in input rows? */
+  @Injection( name = "ACCEPT_FILE_NAMES" )
+  public boolean acceptingFilenames;
+
+  /** The stepname to accept filenames from */
+  @Injection( name = "ACCEPT_FILE_STEP" )
+  public String acceptingStepName;
+
+  /** If receiving input rows, should we pass through existing fields? */
+  @Injection( name = "PASS_THROUGH_FIELDS" )
+  public boolean passingThruFields;
+
+  /** The field in which the filename is placed */
+  @Injection( name = "ACCEPT_FILE_FIELD" )
+  public String acceptingField;
+
+  /** The add filenames to result filenames flag */
+  @Injection( name = "ADD_FILES_TO_RESULT" )
+  public boolean isaddresult;
+
+  @Override
+  public Object clone() {
+    try {
+      BaseFileInputFiles cloned = (BaseFileInputFiles) super.clone();
+      cloned.fileName = Arrays.copyOf( fileName, fileName.length );
+      cloned.fileMask = Arrays.copyOf( fileMask, fileMask.length );
+      cloned.excludeFileMask = Arrays.copyOf( excludeFileMask, excludeFileMask.length );
+      cloned.fileRequired = Arrays.copyOf( fileRequired, fileRequired.length );
+      cloned.includeSubFolders = Arrays.copyOf( includeSubFolders, includeSubFolders.length );
+      return cloned;
+    } catch ( CloneNotSupportedException ex ) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)